core/src/test/scala/io/snappydata/SnappyAnalyticsServiceTest.scala

```diff
@@ -52,6 +54,7 @@ class SnappyAnalyticsServiceTest extends SnappyFunSuite
 		val analyticsRowBuffer = queryMemoryAnalytics(s"APP.$columnTableName")
 		val analyticsColumnBuffer = queryMemoryAnalytics(
 			ColumnFormatRelation.cachedBatchTableName(columnTableName))
+		Thread.sleep(1200)
```

Comment:
See @ymahajan's comment about using waitForCriterion. The method can be found in DistributedTestBase.waitForCriterion. To clarify, this has been a traditional source of many occasional test/precheckin failures where timeout would sometimes be violated and on some machines. WaitForCriterion allows one to evaluate a condition once every specified time-interval till a max timeout period.

A scala friendly version of the same can be added to SnappyFunSuite like:

```scala
  def waitForCriterion(check: () => Boolean, description: String, ms: Long, interval: Long,
      throwOnTimeout: Boolean) {
    val criterion = new WaitCriterion {...}
    DistributedTestBase.waitForCriterion(...)
  }
```